packages/js/eslint-plugin/src/rules/key-name-casing.ts

```diff
@@ -0,0 +1,161 @@
+import { SourceCode } from "eslint"
+import { JSONProperty } from "jsonc-eslint-parser/lib/parser/ast"
+
+import { OpenverseRule } from "../utils/rule-creator"
+
+export type CasingKind = "camelCaseWithDot" | "snake_case_with_dot"
+
+export const allowedCaseOptions: CasingKind[] = [
+  "camelCaseWithDot",
+  "snake_case_with_dot",
+]
+
+const checkersMap = {
+  camelCaseWithDot: isCamelCase,
+  snake_case_with_dot: isSnakeCase,
+}
+
+/**
+ * Checks whether the given string has symbols.
+ */
+function hasSymbols(str: string) {
+  return /[\u0021-\u0023\u0025-\u002c./\u003a-\u0040\u005b-\u005e`\u007b-\u007d]/u.test(
+    str
+  ) // without " ", "$", "-" and "_"
```

Comment:
Because this rule is only used for locale json files, it might be easier to understand this particular check by inverting it to check the characters that are allowed in a locale key. Would that be, `[a-zA-Z._]`, for example? If this rule is only used for locale key strings, and that would allow simplifying aspects of it by making it more exact, then it's worth doing. This regex is a little obscure as it is :sweat_smile: 

In that case, it would also be good to change the name of the rule to `@openverse/locale-key-name-casing`.

Reply:
I copied the code as-is from the key name casing rule, only adding the `.` :) Great suggestions, thank you!